src/server/game/map/index.ts

```diff
@@ -296,14 +303,19 @@ export class Map {
     if (tile.owner) {
       if (!overwrite) return;
       tile.owner?.removeTile(coords);
-      tile.setVisibility(tile.owner.civID, false);
+      if (tile.owner.civID) {
+        tile.setVisibility(tile.owner.civID, false);
+      }
     }
     tile.owner = owner;
-    tile.setVisibility(owner.civID, true);
+    if (owner.civID) {
+      tile.setVisibility(owner.civID, true);
+    }
     owner.addTile(coords);
   }
 
   getCivTile(civID: number, tile: Tile): TileData | null {
+    return tile.getVisibleData(civID);
```

Comment:
This return seems to make the rest of the function unreachable.

Reply:
Stupid mistake - that's debug code. Good catch.